anvil/sample.py

```diff
@@ -103,9 +104,9 @@ def thermalised_state(loaded_model, action, thermalisation) -> torch.Tensor:
     if thermalisation is None:
         return thermalisation
 
-    states, _ = sample_batch(loaded_model, action, thermalisation)
+    states, current_log_density, _ = sample_batch(loaded_model, action, thermalisation)
     log.info(f"Thermalisation: discarded {thermalisation} configurations.")
-    return states[-1]
+    return states[-1], current_log_density
```

Comment:
ok I'm still thinking the best way to do this so don't change yet but I think that we should just make sample_batch return the full stack of both states *and* log density and then this function should index the last item for both because then it becomes canon that whenever we have a phi or a stack of phis, we also have a corresponding log_density

Reply:
i think the symmetry would make it easier to follow